mypy/options.py

```diff
@@ -93,7 +93,7 @@ def __init__(self) -> None:
         self.junit_xml = None  # type: Optional[str]
 
         # Per-module options (raw)
-        self.per_module_options = {}  # type: Dict[str, Dict[str, object]]
+        self.per_module_options = {}  # type: Dict[Pattern[str], Dict[str, object]]
```

Comment:
Yes, there's room for improvement here; I couldn't think of a great way to solve it. How would you determine the order? Does an earlier match override a later match or vice versa? Or would you try to determine the most specific match?